source/blog/2016-07-12-hacking-and-extending-atomic-host.html.md

```diff
@@ -0,0 +1,212 @@
+notice: pkg-add is a preview command and subject to change.
+Need to overlay 1 package onto tree 675583e:
+  strace
+
+Downloading metadata: [===================================================] 100%
+Resolving dependencies... done
+Transaction: 1 packages
+  strace-4.8-11.el7.x86_64 (base)
+Will download: 1 package (271.5 kB)
+
+  Downloading from base: [================================================] 100%
+
+Importing: [==============================================================] 100%
+Checking out tree 675583e... done
+Overlaying... done
+Writing rpmdb... done
+Writing OSTree commit... done
+Copying /etc changes: 34 modified, 8 removed, 44651 added
+Transaction complete; bootconfig swap: no deployment count change: 0
+Freed objects: 26.0 MB
+Added:
+  strace-4.8-11.el7.x86_64
+Run "systemctl reboot" to start a reboot
+-bash-4.2# reboot
+```
+
+Once we're rebooted, we can make use of the new package:
+
+```
+-bash-4.2# strace -V
+strace -- version 4.8
+```
+
+The status output of `rpm-ostree` will now also reflect this layering (which by
+the way received a face lift in this release):
+
+```
+-bash-4.2# rpm-ostree status
+State: idle
+Deployments:
+* centos-atomic-continuous:centos-atomic-host/7/x86_64/devel/continuous
+       Version: 2016.0.169 (2016-07-12 16:26:00)
+    BaseCommit: 675583e7db81b59a08e0bbfb3735bddde9d768f9f4500b845799c7e97ab71091
+        Commit: e0291b49619645d4df95c0924a48932f217b0ed4680656ebd47c1b17640a62ee
+        OSName: centos-atomic-host
+  GPGSignature: (unsigned)
+      Packages: strace
+
+  centos-atomic-continuous:centos-atomic-host/7/x86_64/devel/continuous
+       Version: 2016.0.169 (2016-07-11 21:51:17)
+        Commit: 675583e7db81b59a08e0bbfb3735bddde9d768f9f4500b845799c7e97ab71091
+        OSName: centos-atomic-host
+  GPGSignature: (unsigned)
+```
+
+Not surprisingly, packages that were layered using `pkg-add` can also be later
+removed by invoking the `pkg-remove` command, left here as an exercise to the
+reader.
+
+Under the hood, `rpm-ostree` takes care of many subtleties, including safely
+running RPM `%post` scripts using
+[bubblewrap](https://github.com/projectatomic/bubblewrap) to ensure that the
+OSTree system repo remains uncorrupted, and obeying expected RPM rules (e.g.
+layering a package that conflicts with an existing package will error out).
+
+To summarize, the power that package layering gives you is the ability to
+customize the package set on particular hosts *without having to ship it to all
+hosts*.
+
+Finally, note that package layering is currently in preview mode, which means
+that it is subject to change, though we will do our best to retain backwards
+compatibility with layered deployments created with the v2016.4 release.
+
+### Bonus section: CentOS Atomic Host Continuous (CAHC)
+
+You might have been wondering what this `centos-atomic-continuous` remote was
+about in the command outputs above. Colin Walters, as part of the [CentOS Atomic
```

Comment:
I appreciate the mention but would prefer if we rephrased it like:  "The [CentOS Atomic SIG](https://wiki.centos.org/SpecialInterestGroup/Atomic), has set up an automated job..." or so.  The reason is that I want to get away from a default model where one person "owns" jobs or builds.